### PR TITLE
Add an option to disable windown auth

### PR DESCRIPTION
**Pull request by pakrym**
Issue: https://github.com/aspnet/IISIntegration/issues/15
@Tratcher

### CODE REVIEW

File: src/Microsoft.AspNet.IISPlatformHandler/IISPlatformHandlerOptions.cs [hunk 1]
```diff
@@ -16,6 +16,12 @@ public class IISPlatformHandlerOptions
         public bool AutomaticAuthentication { get; set; } = true;
 
         /// <summary>
+        /// If true authentication middleware will try to authenticate using platform handler windows authentification
+        /// If false authentication middleware won't be added
+        /// </summary>
+        public bool EnableWindowsAuthentication { get; set; } = true;
```
Comment (Tratcher) [inline src/Microsoft.AspNet.IISPlatformHandler/IISPlatformHandlerOptions.cs]: FlowWindowsAuthentication?

File: src/Microsoft.AspNet.IISPlatformHandler/IISPlatformHandlerOptions.cs [hunk 2]
```diff
@@ -16,6 +16,12 @@ public class IISPlatformHandlerOptions
         public bool AutomaticAuthentication { get; set; } = true;
 
         /// <summary>
+        /// If true authentication middleware will try to authenticate using platform handler windows authentification
```
Comment (Tratcher) [inline src/Microsoft.AspNet.IISPlatformHandler/IISPlatformHandlerOptions.cs]: Spelling: `authentification`

File: src/Microsoft.AspNet.IISPlatformHandler/IISPlatformHandlerOptions.cs (hunk 1 above)
Comment (pakrym) [inline src/Microsoft.AspNet.IISPlatformHandler/IISPlatformHandlerOptions.cs]: Flow is a little vague but if you like it ill change

File: src/Microsoft.AspNet.IISPlatformHandler/IISPlatformHandlerMiddleware.cs [hunk 3]
```diff
@@ -49,19 +49,23 @@ public IISPlatformHandlerMiddleware(RequestDelegate next, IISPlatformHandlerOpti
...
 
             UpdateRemoteIp(httpContext);
-
-            var winPrincipal = UpdateUser(httpContext);
-
-            var handler = new AuthenticationHandler(httpContext, _options, winPrincipal);
-            AttachAuthenticationHandler(handler);
-
-            try
+            if (_options.EnableWindowsAuthentication)
```
Comment (Tratcher) [inline src/Microsoft.AspNet.IISPlatformHandler/IISPlatformHandlerMiddleware.cs]: Wasn't updated, doesn't compile.

**Comment by pakrym** [comment]
:up: :date:

**Comment by davidfowl** [comment]
Why is there a merge in the commits? You're going to squash this right?

**Comment by pakrym** [comment]
I edited file in github web and then git pulled, definitely will squash

File: src/Microsoft.AspNet.IISPlatformHandler/IISPlatformHandlerOptions.cs [hunk 4]
```diff
@@ -16,6 +16,12 @@ public class IISPlatformHandlerOptions
         public bool AutomaticAuthentication { get; set; } = true;
 
         /// <summary>
+        /// If true authentification middleware will try to authenticate using platform handler windows authentification
```
Comment (Tratcher) [inline src/Microsoft.AspNet.IISPlatformHandler/IISPlatformHandlerOptions.cs]: Spelling: `authentification`

**Comment by Tratcher** [comment]
:shipit:

**Pull request merged**